app/models/my_registrations/statuses.rb

```diff
@@ -41,94 +30,19 @@ def get_registration_statuses
-    end
-
-    def find_message_by_number(message_nbr)
-      registration_messages.find do |message|
-        message.try(:[], :messageNbr).to_i == message_nbr
-      end.try(:[], :descrlong)
-    end
-
-    def berkeley_terms
-      @berkeley_terms ||= Berkeley::Terms.fetch
-    end
-
-    def registrations
-      @registrations ||= HubEdos::StudentApi::V1::Registrations.new(user_id: @uid).get
-      @registrations.try(:[], :feed).try(:[], 'registrations') || []
-    end
-
-    def student_attributes
-      @student_attributes ||= HubEdos::StudentApi::V1::StudentAttributes.new(user_id: @uid).get
-      @student_attributes.try(:[], :feed).try(:[], 'student').try(:[], 'studentAttributes') || []
-    end
-
-    def positive_service_indicators
-      positive_indicators = []
-      student_attributes.each do |attribute|
-        positive_indicators.push(attribute) if is_positive_service_indicator? attribute
-      end
-      positive_indicators.each do |indicator|
-        check_indicator_dates indicator
-      end
-      positive_indicators
-    end
-
-    def get_terms
-      terms = {}
-      # current, running, and sis_current_term can all potentially be different depending on where we are in the academic year.
-      # So, we grab all of them in case of term transitions.
-      [:current, :running, :sis_current_term, :next, :future].each do |term_method|
-        if (term = berkeley_terms.send term_method)
-          # We need various dates to determine CNP status
-          terms[term_method] = {
-            id: term.campus_solutions_id,
-            name: term.to_english,
-            classesStart: term.classes_start,
-            end: term.end,
-            endDropAdd: term.end_drop_add
-          }
-          terms[term_method] = set_term_flags(terms[term_method])
-        # Often ':future' will be nil, but during Spring terms, it should send back data for the upcoming Fall semester.
-        else
-          terms[term_method] = nil
-        end
-      end
-      terms
-    end
-
-    def set_term_flags(term)
-      current_date = Settings.terms.fake_now || DateTime.now
-      term.merge({
-        # CNP logic dictates that grad/law students are dropped one day AFTER the add/drop deadline.
-        pastAddDrop: term[:endDropAdd] ? current_date > term[:endDropAdd] : nil,
-        # Undergrad students are dropped on the first day of instruction.
-        pastClassesStart: current_date >= term[:classesStart],
-        # All term registration statuses are hidden the day after the term ends.
-        pastEndOfInstruction: current_date > term[:end],
-        # Financial Aid disbursement is used in CNP notification.  This is defined as 9 days before the start of instruction.
-        pastFinancialDisbursement: current_date >= (term[:classesStart] - 9),
-      })
-    end
-
     # Match registration terms with Berkeley::Terms-defined terms.
-    def match_terms(registrations, terms)
+    def match_terms(registrations = [], terms)
```

Comment:
With ordered arguments like this, using the default argument for registrations is impossible unless you also specific a default argument for terms